### DISCUSSION TITLE
How to compress an Image using compress function?

### DISCUSSION THREAD

**Discussion by Gaurav-Harman** [lf-edge/ekuiper#2157] [Q&A]
How can I compress an image using compress function. Also where I can find that saved compressed image. 

For Example: 
I have an image in jpeg and I want to compress the image with extension 7Z as shown below.
![image](https://github.com/lf-edge/ekuiper/assets/141431842/0dc32ad0-a899-4d93-9386-807a1f741b9c)

Also I want image should be compressed and saved in a folder.

**Comment by Rui-Gan** [comment]
Unfortunately, we do not currently have support for compressing images in the 7z format. However, we do offer an image sink feature that enables you to conveniently store images in specific folders.（ https://ekuiper.org/docs/en/latest/guide/sinks/plugin/image.html ）

**Reply by Gaurav-Harman** [reply]
Can we compress the image in any other format? If yes how can we do that. If not than what is the purpose of compress function. How should we use compress function?

**Reply by ngjaying** [reply]
Compress function is typically used to compress part of your payload. For example, the payload like {"a": "something", "b": []byte(0x01, 0x02...)}, you can compress b by  `SELECT compress(b, "zlib")`. If you like, you can use it to compress your image data like `SELECT compress(self, "zlib")`, but that seems not a good idea because the output is no longer a render-able image.

**Reply by Gaurav-Harman** [reply]
How can we save the zlib file which we get as output from compress function.

**Reply by ngjaying** [reply]
File sink is supposed to save result into file system. It does support compression as one of the properties https://ekuiper.org/docs/en/latest/guide/sinks/builtin/file.html#properties.

However, file sink now does not support raw binary format. I am not sure if the lines format can support the raw bytes. Here is an example:

```
{
  "id": "ruleRecord",
  "sql": "SELECT meta(topic) as topic, * FROM mqttDemo",
  "actions": [
    {
      "file": {
        "path": "compressed.gzip",
        "fileType": "lines",
        "rollingCount": 1,
        "rollingNamePattern": "prefix",
        "compression": "gzip",
        "sendSingle": true
      }
    }
  ]
}
```
